### PR TITLE
Update Inline JSX and Type Definitions (1/2)

### PR DESCRIPTION
**Pull request by NickGerleman**
This is a best effort attempt at updating all of the non-snack code examples to TypeScript, adding type definitions to signatures, updating existing type signature documentation to align to TS instead of Flow, and applying a couple formatting changes for consistency with RN's prettier config.

A lot of instances of `removeListener` have been removed in code, and are now removed on the website. I also removed some APIs which should not be called by user code (e.g. functions called by native from the framework).

I also marked a couple of the removed APIs as removed instead of deprecated, and didn't bother to update them.

Finally, we now consistently end function declarations with a semicolon where before some docs did, and others didn't.

### CODE REVIEW

File: docs/flatlist.md [hunk 1]
```diff
@@ -755,8 +764,8 @@ Displays the scroll indicators momentarily.
 
 ### `getNativeScrollRef()`
 
-```jsx
-getNativeScrollRef();
+```tsx
+getNativeScrollRef(): React.ElementRef<ScrollViewComponent>;
```
Comment (NickGerleman) [inline docs/flatlist.md:768]:
```suggestion
getNativeScrollRef(): React.ElementRef<typeof ScrollViewComponent>;
```

**Review by lunaleaps — APPROVED** [review]
💯 !!

File: docs/actionsheetios.md [hunk 2]
```diff
@@ -111,8 +114,12 @@ Dismisses the most upper iOS action sheet presented, if no action sheet is prese
...
 ### `showShareActionSheetWithOptions()`
 
-```jsx
-static showShareActionSheetWithOptions(options, failureCallback, successCallback)
+```tsx
+static showShareActionSheetWithOptions: (
+  options: ShareActionSheetIOSOptions,
+  failureCallback: (error: Error) => void,
+  successCallback: (success: boolean, method: string) => void,
+)
```
Comment (NickGerleman) [inline docs/actionsheetios.md:122]:
```suggestion
);
```

File: docs/appstate.md [hunk 3]
```diff
@@ -215,27 +215,20 @@ Received when the user is not actively interacting with the app. Useful in situa
 
 ### `addEventListener()`
 
-```jsx
-addEventListener(eventType, listener);
+```tsx
+addEventListener(
```
Comment (NickGerleman) [inline docs/appstate.md:219]:
```suggestion
static addEventListener(
```

File: docs/appstate.md [hunk 4]
```diff
@@ -215,27 +215,20 @@ Received when the user is not actively interacting with the app. Useful in situa
...
-
-> **Deprecated.** Use the `remove()` method on the `EventSubscription` returned by [`addEventListener()`](#addeventlistener).
-
 ## Properties
 
 ### `currentState`
 
-```jsx
+```tsx
 AppState.currentState;
```
Comment (NickGerleman) [inline docs/appstate.md:233]:
```suggestion
static currentState: AppStateStatus;
```

File: docs/communication-android.md [hunk 5]
```diff
@@ -55,7 +55,7 @@ class MainActivity : ReactActivity() {
         return object : ReactActivityDelegate(this, mainComponentName) {
             override fun getLaunchOptions(): Bundle {
                 val imageList = arrayListOf("http://foo.com/bar1.png", "http://foo.com/bar2.png")
-                val initialProperties = Bundle().apply { putStringArrayList("images", imageList) }
+                val initialProperties = Bundle().apply {putStringArrayList("images", imageList)}
```
Comment (NickGerleman) [inline docs/communication-android.md:58]:
```suggestion
                val initialProperties = Bundle().apply { putStringArrayList("images", imageList) }
```

File: docs/dimensions.md [hunk 6]
```diff
@@ -7,13 +7,13 @@ import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import con
 
 > [`useWindowDimensions`](usewindowdimensions) is the preferred API for React components. Unlike `Dimensions`, it updates as the window's dimensions update. This works nicely with the React paradigm.
 
-```jsx
+```tsx
 import { Dimensions } from 'react-native';
```
Comment (NickGerleman) [inline docs/dimensions.md:11]:
```suggestion
import {Dimensions} from 'react-native';
```

File: docs/native-components-android.md [hunk 7]
```diff
@@ -815,7 +815,7 @@ public class MyPackage implements ReactPackage {
 
 I. Start with custom View manager:
 
-```jsx title="MyViewManager.jsx"
+```tsx title="MyViewManager.jsx"
```
Comment (NickGerleman) [inline docs/native-components-android.md:818]:
```suggestion
```tsx title="MyViewManager.tsx"
```

File: docs/native-components-android.md [hunk 8]
```diff
@@ -824,7 +824,7 @@ export const MyViewManager =
 
 II. Then implement custom View calling the `create` method:
 
-```jsx title="MyView.jsx"
+```tsx title="MyView.jsx"
```
Comment (NickGerleman) [inline docs/native-components-android.md:827]:
```suggestion
```tsx title="MyView.tsx"
```

File: docs/native-modules-android.md [hunk 9]
```diff
@@ -929,7 +929,7 @@ Let's implement a basic image picker to demonstrate this. The image picker will
 <Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
 <TabItem value="java">
 
-```jsx
+```tsx
```
Comment (NickGerleman) [inline docs/native-modules-android.md:932]:
```suggestion
```kt
```

File: docs/native-modules-android.md [hunk 10]
```diff
@@ -1022,14 +1022,14 @@ class ImagePickerModule(reactContext: ReactApplicationContext) :
                 intent: Intent?
             ) {
                 if (requestCode == IMAGE_PICKER_REQUEST) {
-                    pickerPromise?.let { promise ->
+                    pickerPromise?.let {promise ->
```
Comment (NickGerleman) [inline docs/native-modules-android.md:1025]:
```suggestion
                    pickerPromise?.let { promise ->
```

File: docs/native-modules-android.md [hunk 11]
```diff
@@ -1022,14 +1022,14 @@ class ImagePickerModule(reactContext: ReactApplicationContext) :
...
                         when (resultCode) {
                             Activity.RESULT_CANCELED ->
                                 promise.reject(E_PICKER_CANCELLED, "Image picker was cancelled")
                             Activity.RESULT_OK -> {
                                 val uri = intent?.data
 
-                                uri?.let { promise.resolve(uri.toString()) }
+                                uri?.let {promise.resolve(uri.toString())}
```
Comment (NickGerleman) [inline docs/native-modules-android.md:1032]:
```suggestion
                                uri?.let { promise.resolve(uri.toString())}
```

File: docs/native-modules-android.md [hunk 12]
```diff
@@ -1058,7 +1058,7 @@ class ImagePickerModule(reactContext: ReactApplicationContext) :
         pickerPromise = promise
 
         try {
-            val galleryIntent = Intent(Intent.ACTION_PICK).apply { type = "image\/*" }
+            val galleryIntent = Intent(Intent.ACTION_PICK).apply {type = "image\/*"}
```
Comment (NickGerleman) [inline docs/native-modules-android.md:1061]:
```suggestion
            val galleryIntent = Intent(Intent.ACTION_PICK).apply { type = "image\/*" }
```

File: docs/native-modules-ios.md [hunk 13]
```diff
@@ -280,14 +280,14 @@ A native module can export constants by overriding the native method `constantsT
 ```objectivec
 - (NSDictionary *)constantsToExport
 {
- return @{ @"DEFAULT_EVENT_NAME": @"New Event" };
+ return @{@"DEFAULT_EVENT_NAME": @"New Event"};
```
Comment (NickGerleman) [inline docs/native-modules-ios.md:283]:
```suggestion
 return @{ @"DEFAULT_EVENT_NAME": @"New Event" };
```

File: docs/native-modules-ios.md [hunk 14]
```diff
@@ -280,14 +280,14 @@ A native module can export constants by overriding the native method `constantsT
...
 }
 ```
 
 The constant can then be accessed by invoking `getConstants()` on the native module in JS like so:
 
 ```objectivec
```
Comment (NickGerleman) [inline docs/native-modules-ios.md:289]:
```suggestion
```tsx
```

**Pull request merged**